src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -276,30 +277,7 @@ RelayEventExtendNames(Node *parseTree, char *schemaName, uint64 shardId)
 			break;
 		}
 
-		case T_CreateForeignServerStmt:
-		{
-			CreateForeignServerStmt *serverStmt = (CreateForeignServerStmt *) parseTree;
-			char **serverName = &(serverStmt->servername);
-
-			AppendShardIdToName(serverName, shardId);
-			break;
-		}
-
 		case T_CreateForeignTableStmt:
-		{
```

Comment:
Hmm, why do we remove these two? 

I can see that when we drop distributed table support for foreign tables, we won't be sending ` SELECT worker_apply_shard_ddl_command (102010, 'ALTER FOREIGN TABLE XXXX')` types of commands to the workers, and these would not be useful. But, now we should be sending such commands, no? I'm surprised that no test failed after this.

As we discussed earlier, we should make sure that all DDLs happening on `foreign tables` should do `ALTER FOREIGN TABLE `..., not `ALTER TABLE`.

We should not rely Postgres to treat both commands the same.

Reply:
This is not about foreign tables, it's about servers. We don't append shard_ids, or hashes to server names anymore. All servers are created on the workers with the original name.

Reply:
I mostly asked regarding `T_CreateForeignTableStmt`. We are now casting `CreateForeignTableStmt` to `CreateStmt`. But, that seems fine as they share the same `base`.

PG does the same: https://github.com/postgres/postgres/blob/4168a4745492cd54a0ffffc271b452525ef4dc60/src/backend/tcop/utility.c#L1133-L1142

Also, in the past how/why we didn't have `AppendShardIdToName(relationName, shardId);` for `CreateForeignTableStmt`. Do you know why?

Reply:
I'm not sure I've understood your question, correct me if I get it wrong:
We have that for `CreateStmt`.
`CreateForeignTableStmt` falls through there.